substrate/frame/vesting/src/lib.rs

```diff
@@ -378,6 +378,25 @@ pub mod pallet {
 			Self::do_vested_transfer(source, target, schedule)
 		}
 
+		/// Force remove a vesting schedule
+		///
+		/// The dispatch origin for this call must be _Root_.
+		///
+		/// - `target`: An account that has a vesting schedule
+		/// - `schedule_index`: The vesting schedule index that should be removed
+		#[pallet::weight(
+		T::WeightInfo::force_remove_vesting_schedule(MaxLocksOf::<T>::get(), T::MAX_VESTING_SCHEDULES)
+		)]
+		pub fn force_remove_vesting_schedule(
+			origin: OriginFor<T>,
+			target: <T::Lookup as StaticLookup>::Source,
+			schedule_index: u32,
+		) -> DispatchResult {
+			ensure_root(origin)?;
+			let who = T::Lookup::lookup(target)?;
+			Self::remove_vesting_schedule(&who, schedule_index)
+		}
```

Comment:
We could refund the weight for traversing the schedules here by decoding the length of the map entry under `who` before calling `remove_vesting_schedule`. For this I think the function would need to return `DispatchResultWithPostInfo`. Example syntax [here](https://github.com/paritytech/polkadot-sdk/blob/91851951856b8effe627fb1d151fe336a51eef2d/cumulus/pallets/collator-selection/src/lib.rs#L422) and [here](https://github.com/paritytech/polkadot-sdk/blob/91851951856b8effe627fb1d151fe336a51eef2d/cumulus/pallets/collator-selection/src/lib.rs#L458).

Reply:
Ok, that's done.